recovery-init

```diff
@@ -34,39 +104,45 @@ mkdir /dev/pts
 mount -a
 cat /proc/mounts > /etc/mtab
 
+mkdir /config
+mount -t configfs none /config
+
 #Run mdev
 echo "Running mdev..."
 mdev -s
+
+
 #Set mdev as hotplug manager
 echo /bin/mdev > /proc/sys/kernel/hotplug
 
-write()
-{
-	echo -n "$2" > $1
-}
 
 # Minimize power consumption by lowering display brightness to minimum
 write $DISPLAY_BRIGHTNESS_PATH $DISPLAY_BRIGHTNESS
 
-write /sys/class/android_usb/android0/enable 0
-write /sys/class/android_usb/android0/idVendor 2931
-write /sys/class/android_usb/android0/idProduct 0A06
-write /sys/class/android_usb/android0/functions rndis,mass_storage
-write /sys/class/android_usb/android0/f_rndis/ethaddr 00:01:02:03:04:05
-write /sys/class/android_usb/android0/enable 1
-write /sys/class/android_usb/android0/iManufacturer Jolla
-write /sys/class/android_usb/android0/iProduct Recovery
-write /sys/class/android_usb/android0/iSerial 123456789
-write /sys/class/android_usb/android0/enable 1
+
+
+usb_setup "Jolla Debug setting up"
 
 ip link set rndis0 up
-ipaddr add 10.42.66.66/29 broadcast 10.42.66.255 dev rndis0
-ipaddr add 192.168.2.15/24 broadcast 192.168.2.255 dev rndis0 label rndis0:0
+if ipaddr add 10.42.66.66/29 broadcast 10.42.66.255 dev rndis0 ; then
+    USB_IFACE=rndis0
+elif ipaddr add 10.42.66.66/29 broadcast 10.42.66.255 dev usb0 ; then
+    USB_IFACE=usb0
+else
+    usb_info "Jolla Debug: ERROR: could not setup USB as usb0 or rndis0"
+    dmesg
+    sleep 60 # plenty long enough to check usb on host
+    reboot -f
+fi
+ipaddr add 192.168.2.15/24 broadcast 192.168.2.255 dev $USB_IFACE label rndis0:0
+
 udhcpd
 
 echo V > /dev/watchdog
 
-yamui -t "RECOVERY: Connect USB cable and open telnet to address 10.42.66.66" &
+
+usb_info "Jolla Debug telnet on $USB_IFACE $LOCAL_IP - also running udhcpd"
+yamui -t "RECOVERY: telnet on $USB_IFACE $LOCAL_IP" &
```

Comment:
Please keep yamui info same as it was before -> "RECOVERY: Connect USB cable and open telnet to address $USB_IFACE $LOCAL_IP"